lib/puppet/module.rb

```diff
@@ -52,6 +53,30 @@ def self.is_module_namespaced_name?(name)
     return false
   end
 
+  # @api private
+  def self.parse_range(range, strict)
+    @parse_range_method ||= SemanticPuppet::VersionRange.method(:parse)
+    if @parse_range_method.arity == 1
```

Comment:
this is the coolest block of ruby I read this month. I didn't know that ruby is able to tell you the amount of arguments.

---

Thanks for quickly providing such an awesome fix!